lib/actions/apiV2.js

```diff
@@ -920,11 +923,28 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
           routingError,
           {
             rewritePayload: (response, dispatch, getState) => {
-              const withCollapsedShortNames =
-                response.data?.plan?.itineraries?.map((itin) => ({
+              const itineraries = response.data?.plan?.itineraries
+              const activeModeStrings = activeModes.map(
+                (am) => SIMPLIFICATIONS[am.mode]
+              )
+              let filteredItineraries
+
+              if (strictModes) {
+                filteredItineraries = itineraries.filter((itin) =>
+                  itin.legs.some((leg) =>
+                    activeModeStrings.includes(SIMPLIFICATIONS[leg.mode])
+                  )
+                )
+              } else {
+                filteredItineraries = itineraries
+              }
```

Comment:
Wouldn't it be easier to just always run the filter and return true if `strictModes`?

Reply:
easier sure, but less efficient.